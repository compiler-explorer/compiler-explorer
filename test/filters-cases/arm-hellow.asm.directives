.Ltext0:
main:
.LFB31:
        @ args = 0, pretend = 0, frame = 0
        @ frame_needed = 0, uses_anonymous_args = 0
.LVL0:
        push    {r3, lr}
.LCFI0:
.LBB6:
.LBB7:
        movs    r0, #1
        movw    r1, #:lower16:.LC0
        movt    r1, #:upper16:.LC0
        bl      __printf_chk
.LVL1:
.LBE7:
.LBE6:
.LBB8:
.LBB9:
        movs    r0, #1
        movw    r1, #:lower16:.LC1
        movt    r1, #:upper16:.LC1
        bl      __printf_chk
.LBE9:
.LBE8:
        movs    r0, #0
        pop     {r3, pc}
.LFE31:
.LC0:
        .ascii  "Hello world\000"
.LC1:
        .ascii  "moo\012\000"
.Letext0:
.Ldebug_info0:
.Ldebug_abbrev0:
.Ldebug_loc0:
.LLST0:
.Ldebug_ranges0:
.Ldebug_line0:
.LASF22:
.LASF9:
.LASF30:
.LASF25:
.LASF5:
.LASF8:
.LASF50:
.LASF35:
.LASF19:
.LASF14:
.LASF21:
.LASF26:
.LASF16:
.LASF6:
.LASF34:
.LASF11:
.LASF53:
.LASF31:
.LASF47:
.LASF46:
.LASF43:
.LASF1:
.LASF4:
.LASF7:
.LASF0:
.LASF44:
.LASF33:
.LASF18:
.LASF42:
.LASF15:
.LASF2:
.LASF13:
.LASF55:
.LASF45:
.LASF36:
.LASF37:
.LASF38:
.LASF39:
.LASF40:
.LASF54:
.LASF3:
.LASF20:
.LASF12:
.LASF10:
.LASF27:
.LASF24:
.LASF48:
.LASF29:
.LASF41:
.LASF17:
.LASF32:
.LASF51:
.LASF23:
.LASF28:
.LASF49:
.LASF52:
