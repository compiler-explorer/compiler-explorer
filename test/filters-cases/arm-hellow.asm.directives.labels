main:
        @ args = 0, pretend = 0, frame = 0
        @ frame_needed = 0, uses_anonymous_args = 0
        push    {r3, lr}
        movs    r0, #1
        movw    r1, #:lower16:.LC0
        movt    r1, #:upper16:.LC0
        bl      __printf_chk
        movs    r0, #1
        movw    r1, #:lower16:.LC1
        movt    r1, #:upper16:.LC1
        bl      __printf_chk
        movs    r0, #0
        pop     {r3, pc}
.LC0:
        .ascii  "Hello world\000"
.LC1:
        .ascii  "moo\012\000"
