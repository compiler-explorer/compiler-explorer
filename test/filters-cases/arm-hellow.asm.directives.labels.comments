main:
        push    {r3, lr}
        movs    r0, #1
        movw    r1, #:lower16:.LC0
        movt    r1, #:upper16:.LC0
        bl      __printf_chk
        movs    r0, #1
        movw    r1, #:lower16:.LC1
        movt    r1, #:upper16:.LC1
        bl      __printf_chk
        movs    r0, #0
        pop     {r3, pc}
.LC0:
        .ascii  "Hello world\000"
.LC1:
        .ascii  "moo\012\000"
