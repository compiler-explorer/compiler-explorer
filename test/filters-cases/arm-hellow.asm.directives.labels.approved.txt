{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": "main:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        @ args = 0, pretend = 0, frame = 0"
    },
    {
      "labels": [],
      "source": null,
      "text": "        @ frame_needed = 0, uses_anonymous_args = 0"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 2
      },
      "text": "        push    {r3, lr}"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movs    r0, #1"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movw    r1, #:lower16:.LC0"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movt    r1, #:upper16:.LC0"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        bl      __printf_chk"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movs    r0, #1"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movw    r1, #:lower16:.LC1"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        movt    r1, #:upper16:.LC1"
    },
    {
      "labels": [],
      "source": {
        "file": "/usr/arm-linux-gnueabi/include/bits/stdio2.h",
        "line": 105
      },
      "text": "        bl      __printf_chk"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 5
      },
      "text": "        movs    r0, #0"
    },
    {
      "labels": [],
      "source": {
        "file": null,
        "line": 5
      },
      "text": "        pop     {r3, pc}"
    },
    {
      "labels": [],
      "source": null,
      "text": ".LC0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .ascii  \"Hello world\\000\""
    },
    {
      "labels": [],
      "source": null,
      "text": ".LC1:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .ascii  \"moo\\012\\000\""
    }
  ],
  "labelDefinitions": {
    ".LC0": 15,
    ".LC1": 17,
    "main": 1
  }
}
